src/graph.py

```diff
@@ -0,0 +1,162 @@
+"""Implement a class for a graph data structure."""
+
+
+class Graph(object):
+    """Graph class."""
+
+    def __init__(self):
+        """Initialize a graph."""
+        self._graph = {}
+
+    def nodes(self):
+        """Return a list of all nodes in graph."""
+        return list(self._graph.keys())
+
+    def edges(self):
+        """Return a list of edges in graph."""
+        edges = []
+        for key in self._graph:
+            for i in self._graph[key]:
+                edges.append((key, i, self._graph[key][i]))
+        return edges
+
+    def add_node(self, val):
+        """Add a node with value of val to graph."""
+        self._graph[val] = {}
+
+    def add_edge(self, val1, val2, weight=0):
+        """Add a new edge to graph between val1 & val2 as well as add vals."""
+        if val1 in self._graph and val2 in self._graph:
+            if val2 not in self._graph[val1]:
+                self._graph[val1][val2] = weight
+        elif val1 in self._graph and val2 not in self._graph:
+            self._graph[val2] = {}
```

Comment:
You're adding a new node here. You have a method whose job it is to add new nodes. Don't forget that